convert2rhel/pkghandler.py

```diff
@@ -612,9 +612,12 @@ def replace_non_rhel_installed_kernel(version):
 
     pkg = "kernel-%s" % version
 
+    # For downloading the RHEL kernel we need to use the RHEL repositories. Those are available either through the
+    # attached subscription (submgr_enabled_repos) or through custom repositories (tool_opts.enablerepo).
+    repos_to_enable = system_info.submgr_enabled_repos if not tool_opts.disable_submgr else tool_opts.enablerepo
```

Comment:
FYI, this is 1:1 with https://github.com/oamg/convert2rhel/blob/main/convert2rhel/pkghandler.py#L114